subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/versionupgrade/FindOpenPullRequest.java

```diff
@@ -0,0 +1,40 @@
+package org.shipkit.internal.gradle.versionupgrade;
+
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.json.simple.DeserializationException;
+import org.json.simple.JsonArray;
+import org.json.simple.JsonObject;
+import org.json.simple.Jsoner;
+import org.shipkit.internal.util.GitHubApi;
+
+import java.io.IOException;
+
+class FindOpenPullRequest {
+
+    private static final Logger LOG = Logging.getLogger(FindOpenPullRequest.class);
+
+    public String findOpenPullRequest(FindOpenPullRequestTask task) throws IOException, DeserializationException {
+        return findOpenPullRequest(task, new GitHubApi(task.getGitHubApiUrl(), task.getAuthToken()));
+    }
+
+    public String findOpenPullRequest(FindOpenPullRequestTask task, GitHubApi gitHubApi) throws IOException, DeserializationException {
+        String response = gitHubApi.get("/repos/" + task.getUpstreamRepositoryName() + "/pulls?state=open");
+
+        JsonArray pullRequests = Jsoner.deserialize(response, new JsonArray());
+
+        for (Object pullRequest : pullRequests) {
+            JsonObject head = (JsonObject) ((JsonObject) pullRequest).get("head");
+            String branchName = head.getString("ref");
+            if (branchName.matches(task.getVersionBranchRegex())) {
+                LOG.lifecycle("  Found an open pull request with version upgrade on branch {}", branchName);
+                return head.getString("ref");
+            }
+        }
+
+        LOG.lifecycle("  No open pull request with version upgrade found.");
```

Comment:
This message may be misleading to the user. Is it a problem that the pull request was not found? If the build fails and the user investigates the log he might think that it has something to do with that. I suggest to write something like (please tweak it :):

```
New pull request will be opened because we didn't find an existing PR to reuse
```